docs/data/rpc/api-reference/methods/getLedgerEntries.mdx

```diff
@@ -48,291 +147,179 @@ print(
-        "AAAABgAAAAAAAAABn4yVRAXOeXa1t1POzdT8tEcNP55alAOrUOq9vJIRJeIAAAAUAAAAAQAAABMAAAAA5DNtbckOGVRsNVb8L7X/lIhAOy2o5G6GkLKXvc7W8foAAAAA"
-    )
-)
-# OUTPUT: AAAAB+QzbW3JDhlUbDVW/C+1/5SIQDstqORuhpCyl73O1vH6
+def get_ledger_key_wasm_id(
+  # received from getLedgerEntries and decoded
+  contract_data: xdr.ContractDataEntry
+) -> xdr.LedgerKey:
+  # First, we dig the wasm_id hash out of the xdr we received from RPC
+  wasm_hash = contract_data.val.instance.executable.wasm_hash
+
+  # Now, we can create the `LedgerKey` as we've done in previous examples
+  ledger_key = xdr.LedgerKey(
+    type=xdr.LedgerEntryType.CONTRACT_CODE,
+    contract_code=xdr.LedgerKeyContractCode(
+      hash=wasm_hash
+    ),
+  )
+  return ledger_key
 ```
 
-##### JavaScript
-
-```javascript
+```typescript
 import { xdr } from "@stellar/stellar-sdk";
 
-function getLedgerKeyWasmId(contractCodeLedgerEntryData) {
-  const entry = xdr.LedgerEntryData.fromXDR(
-    contractCodeLedgerEntryData,
-    "base64",
-  );
-
-  const wasmHash = entry
-    .contractData()
-    .val()
-    .instance()
-    .executable()
-    .wasmHash();
+function getLedgerKeyWasmId(
+  contractData: xdr.ContractDataEntry,
+): xdr.LedgerKey {
+  const wasmHash = contractData.val().instance().executable().wasmHash();
 
-  let ledgerKey = xdr.LedgerKey.contractCode(
+  return xdr.LedgerKey.contractCode(
     new xdr.LedgerKeyContractCode({
       hash: wasmHash,
     }),
   );
-
-  return ledgerKey.toXDR("base64");
 }
+```
 
-console.log(
-  getLedgerKeyWasmId(
-    "AAAABgAAAAAAAAABn4yVRAXOeXa1t1POzdT8tEcNP55alAOrUOq9vJIRJeIAAAAUAAAAAQAAABMAAAAA5DNtbckOGVRsNVb8L7X/lIhAOy2o5G6GkLKXvc7W8foAAAAA",
-  ),
-);
-// OUTPUT: AAAAB+QzbW3JDhlUbDVW/C+1/5SIQDstqORuhpCyl73O1vH6
+</CodeExample>
+
+Now, finally we have a `LedgerKey` that correspond to the Wasm byte-code that has been deployed under the `contractId` we started out with so very long ago. This `LedgerKey` can be used in a final request to `getLedgerEntries`. In that response we will get a `LedgerEntryData` corresponding to a `ContractCodeEntry` which will contain the actual, deployed, real-life contract byte-code:
+
+```typescript
+const theHashData: xdr.ContractDataEntry = await getLedgerEntries(
+  getLedgerKeyContractCode("C..."),
+).entries[0].contractData();
+
+const theCode: Buffer = await getLedgerEntries(getLedgerKeyWasmId(theHashData))
+  .entries[0].contractCode()
+  .code();
+```
+
+## Actually fetching the ledger entry data
+
+Once we've learned to _build_ and _parse_ these (which we've done above at length), the process for actually fetching them is always identical. If you know the type of key you fetched, you apply the accessor method accordingly once you've received them from the `getLedgerEntries` method:
+
+```typescript
+const s = new Server("https://soroban-testnet.stellar.org");
+
+// assume key1 is an account, key2 is a trustline, and key3 is contract data
+const keys = await s.getLedgerEntries(key1, key2, key3);
```

Comment:
`keys` does not seem like an appropriate variable name because it actually contains the ledger entries. Maybe `response` would be a better name.

Reply:
Lol oopsie, good callout